### ISSUE TITLE
[stable/kubernetes-dashboard] cannot create in namespace other than kube-system

### ISSUE THREAD

**Issue by kartoch** [helm/charts#9776]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Version of Helm and Kubernetes**:

```
$ helm version --tiller-namespace tiller-system
Client: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
$ kubectl get nodes
NAME      STATUS    AGE       VERSION
myk8s-0   Ready     1d        v1.12.3
myk8s-1   Ready     1d        v1.12.3
myk8s-2   Ready     1d        v1.12.3
```

**Which chart**: stable/kubernetes-dashboard

**What happened**:

```
$ helm install --tiller-namespace tiller-system stable/kubernetes-dashboard --set "rbac.create=true" --namespace kube-system
```
**CrashLoopBackOff**

**Logs pod:***

```
2018/12/06 13:56:17 Synchronizer kubernetes-dashboard-key-holder-kube-system exited with error: kubernetes-dashboard-key-holder-kube-system watch ended with timeout
2018/12/06 13:56:18 Storing encryption key in a secret
panic: secrets is forbidden: User "system:serviceaccount:default:halting-billygoat-kubernetes-dashboard" cannot create resource "secrets" in API group "" in the namespace "kube-system"

goroutine 1 [running]:
github.com/kubernetes/dashboard/src/app/backend/auth/jwe.(*rsaKeyHolder).init(0xc420408c40)
	/home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/auth/jwe/keyholder.go:131 +0x2d3
github.com/kubernetes/dashboard/src/app/backend/auth/jwe.NewRSAKeyHolder(0x1b13180, 0xc4204163c0, 0xc4204163c0, 0x12e0aca)
	/home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/auth/jwe/keyholder.go:170 +0x83
main.initAuthManager(0x1b12400, 0xc4200d6360, 0xc42028dc68, 0x1)
	/home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/dashboard.go:185 +0x12f
main.main()
	/home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/dashboard.go:103 +0x28c

```

**What you expected to happen**:

No **CarshLoopBackOff* if namespace not "kube-system"

**How to reproduce it** (as minimally and precisely as possible):

(tiller is installed in tiller-system namespace)

```
$ helm install --tiller-namespace tiller-system stable/kubernetes-dashboard --set "rbac.create=true" --namespace kube-system
```

**Comment by yuanpli** [comment]
Faced the same issue. :(

**Comment by nsaboo** [comment]
Facing the same issue :(

**Comment by v-yarotsky** [comment]
It looks like https://github.com/kubernetes/dashboard/pull/3496 added support for the `--namespace` argument, which specifies where the secret gets created.
I installed the chart with the following values:
```
extraArgs:
- --namespace=my-namespace
image:
  repository: kubernetesdashboarddev/kubernetes-dashboard-amd64
  tag: b15d88cf71755b234aed29c1fe5503be59d977b9
  pullPolicy: IfNotPresent
```

Now I'm getting a different error :|

```
2019/01/11 20:14:57 Storing encryption key in a secret
panic: secrets is forbidden: User "system:anonymous" cannot create secrets in the namespace "kubernetes-dashboard"

goroutine 1 [running]:
github.com/kubernetes/dashboard/src/app/backend/auth/jwe.(*rsaKeyHolder).init(0xc00043bc60)
        /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/auth/jwe/keyholder.go:132 +0x356
github.com/kubernetes/dashboard/src/app/backend/auth/jwe.NewRSAKeyHolder(0x15a07a0, 0xc000355320, 0xc000355320, 0x13cf2c0)
        /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/auth/jwe/keyholder.go:171 +0x5f
main.initAuthManager(0x159f4a0, 0xc00011a1e0, 0xc000401c98, 0x1)
        /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/dashboard.go:189 +0x131
main.main()
        /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/dashboard.go:108 +0x289
```

for some reason the k8s api client used in the dashboard codebase tries to use the "system:anonymous" service account...

**Comment by darmenliu** [comment]
same issues for me on windows 7

**Comment by danbeaulieu** [comment]
I was able to reproduce error seen by @v-yarotsky. @floreks any ideas?

**Comment by floreks** [comment]
@v-yarotsky @darmenliu @danbeaulieu we've added support for different namespace deployment to the head image which can be used at your own risk. We do not guarantee that it will be stable or that it will work. It is built and deployed after every master build, so sometimes it might crash. The mentioned issue is known and it will be fixed by this PR (https://github.com/kubernetes/dashboard/pull/3522).

As for the original issue:
> system:serviceaccount:default:halting-billygoat-kubernetes-dashboard

This is different SA than the one we are creating and lacks the required privileges needed to start Dashboard.

**Comment by monotek** [comment]
This can't be solved via the chart. Needs a new dashboard release.
See also: https://github.com/kubernetes/dashboard/pull/3197

**Comment by michaelzion** [comment]
Facing the same issue.
Trying to deploy the dashboard in a namespace called 'monitor', and failing to do so.

**Comment by nicroto** [comment]
Still an issue. +1 on supporting other namespaces.

**Comment by bitsofinfo** [comment]
same

**Comment by FrenchBen** [comment]
@nicroto did you try the latest build: https://github.com/helm/charts/issues/9776#issuecomment-456186972 ?

**Comment by nicroto** [comment]
@FrenchBen I have tested the latest at the time of my previous comment.
Don’t have the time to try again ATM. Perhaps someone else can verify it’s fixed?

**Comment by macalinao** [comment]
any updates?

**Comment by monotek** [comment]
Maybe its worth to give kube web view a try ;-)

https://github.com/hjacobs/kube-web-view

**Comment by kfox1111** [comment]
Its fixed in 2.0. There isnt a final release yet but I hear it should be soon.
